### PR TITLE
Use dor-services-client for object registration and opening new versions

### LINKED ISSUE #228: Use dor-services-client for registration
https://github.com/sul-dlss/argo/issues/1212

### PR DESCRIPTION
**Pull request by mjgiarlo**
Fixes #228 

Also:
* Configure TravisCI build to use docker-compose instead of recreating it
* Add dor-services-app to the docker-compose config
* Upgrade dor-services to 6.1.x 
* Work around underlying changes to Dor::Editable
* Bump to latest dor-services-client and dor-services to pick up new patches (needed because of bugs discovered doing this work)
* Remove deprecation warnings and use client for opening new versions

### CODE REVIEW

File: app/models/hydrus/generic_object.rb [hunk 1]
```diff
@@ -294,17 +294,17 @@ def self.stanford_terms_of_use
...
+    Dor::Services::Client.objects.register(params: params)
   end
 
   # Returns a hash of info needed to register a Dor object.
   def self.dor_registration_params(user_string, obj_typ, apo_pid)
     {
       object_type: obj_typ,
       admin_policy: apo_pid,
-      source_id: { 'Hydrus' => "#{obj_typ}-#{user_string}-#{HyTime.now_datetime_full}" },
+      source_id: "Hydrus:#{obj_typ}-#{user_string}-#{HyTime.now_datetime_full}",
```
Comment (mjgiarlo) [inline app/models/hydrus/generic_object.rb]: The param changes on lines 305 and 307 are due to the switch above on line 297, which behind the scenes now uses `Dor::RegistrationService#create_from_request` whereas it used to use `Dor::RegistrationService#register_object`. And these methods have different expectations around the args they receive.

File: app/models/hydrus/admin_policy_object.rb [hunk 2]
```diff
@@ -81,8 +81,8 @@ def self.create(user)
     # Create the object, with the correct model.
     dconf = Dor::Config.hydrus
     args = [user, 'adminPolicy', dconf.ur_apo_druid]
-    apo  = Hydrus::GenericObject.register_dor_object(*args)
-    apo  = Hydrus::AdminPolicyObject.find(apo.pid)
+    response = Hydrus::GenericObject.register_dor_object(*args)
```
Comment (mjgiarlo) [inline app/models/hydrus/admin_policy_object.rb]: The client returns a hash, not a repository object.

File: app/models/hydrus/collection.rb [hunk 3]
```diff
@@ -266,7 +266,7 @@ def refresh_titles
     apo.identityMetadata.objectLabel = apt
     apo.title                        = apt
     apo.label                        = apt
-    apo.datastreams['DC'].content    = apo.generate_dublin_core.to_s
+    apo.datastreams['DC'].content    = Dor::DublinCoreService.new(apo).to_xml
```
Comment (mjgiarlo) [inline app/models/hydrus/collection.rb]: Remove a deprecation warning.

File: app/models/hydrus/rights_metadata_ds.rb [hunk 4]
```diff
@@ -67,6 +67,14 @@ def remove_license
     remove_nodes_by_xpath(q)
   end
 
+  # Hydrus uses customized rights metadata, but also uses models that include
+  # Dor::Editable. In dor-services 6.x, these methods moved out of the
+  # Dor::Editable module and moved into a datastream which Hydrus does not use.
+  # Stub out these methods to get the test suite running
+  [:default_rights, :use_license].each do |missing_method_name|
```
Comment (mjgiarlo) [inline app/models/hydrus/rights_metadata_ds.rb]: See comment on lines 70-73. Suggestions welcome!

File: config/application.rb [hunk 5]
```diff
@@ -57,6 +57,19 @@ class Application < Rails::Application
...
+    # DEPRECATION WARNING: Currently, Active Record suppresses errors raised
+    # within `after_rollback`/`after_commit` callbacks and only print them to
+    # the logs. In the next version, these errors will no longer be suppressed.
+    # Instead, the errors will propagate normally just like in other Active
+    # Record callbacks.
+    #
+    # You can opt into the new behavior and remove this warning by setting:
+    #
+    # config.active_record.raise_in_transactional_callbacks = true
+    config.active_record.raise_in_transactional_callbacks = true
```
Comment (mjgiarlo) [inline config/application.rb]: Remove a deprecation warning. See lines 61-71.

File: spec/features/contact_spec.rb [hunk 6]
```diff
@@ -9,7 +9,7 @@
     @message_textarea  = 'textarea#message'
     @message           = 'Hello, this is is cool.'
     @send_button_text  = 'Send'
-    HydrusMailer.stub_chain(:contact_message, :deliver_now)
+    allow(HydrusMailer).to receive_message_chain(:contact_message, :deliver_now)
```
Comment (mjgiarlo) [inline spec/features/contact_spec.rb]: Remove a deprecation warning from RSpec.

File: spec/models/hydrus/collection_spec.rb [hunk 7]
```diff
@@ -70,7 +70,7 @@
       }
       tests.each do |meth, predicate|
         allow(@hc).to receive(predicate).and_return(false)
-        expect { @hc.send(meth) }.to raise_error
+        expect { @hc.send(meth) }.to raise_error(RuntimeError)
```
Comment (mjgiarlo) [inline spec/models/hydrus/collection_spec.rb]: Remove an RSpec warning about `raise_error` w/o an argument.

File: spec/support/rubydora_transactions_monkeypatch.rb [hunk 8]
```diff
@@ -32,7 +32,6 @@ def rollback_fixtures(fixtures)
           repository.ingest(pid: p, file: foxml)
           $fixture_solr_cache ||= {}
           $fixture_solr_cache[p] ||= begin
-            puts " indexing and caching #{p}"
```
Comment (mjgiarlo) [inline spec/support/rubydora_transactions_monkeypatch.rb]: Remove noise from the test suite.

**Comment by jcoyne** [comment]
I bet the difference is here:
https://github.com/sul-dlss/hydrus/blob/2b24c3465f4fde0c9e57b7f45163b74443a28ecb/app/models/hydrus/admin_policy_object.rb#L27
https://github.com/sul-dlss/dor-services/blob/master/lib/dor/models/concerns/rightsable.rb#L8

File: app/models/hydrus/admin_policy_object.rb [hunk 9]
```diff
@@ -95,8 +95,6 @@ def self.create(user)
     %w[sdr:developer sdr:service-manager sdr:metadata-staff].each do |group|
       rmd.add_group_with_role(group, 'dor-apo-manager')
     end
-    # Create defaultObjectRights datastream ... by mentioning it.
```
Comment (mjgiarlo) [inline app/models/hydrus/admin_policy_object.rb]: The datastream already exists without this code.

**Pull request merged**